waitress/server.py

```diff
@@ -42,27 +42,95 @@ def create_server(application,
             'to return a WSGI app within your application.'
             )
     adj = Adjustments(**kw)
+
+    if map is None: # pragma: nocover
+        map = {}
+
+    dispatcher = _dispatcher
+    if dispatcher is None:
+        dispatcher = ThreadedTaskDispatcher()
+        dispatcher.set_thread_count(adj.threads)
+
     if adj.unix_socket and hasattr(socket, 'AF_UNIX'):
-        cls = UnixWSGIServer
-    else:
-        cls = TcpWSGIServer
-    return cls(application, map, _start, _sock, _dispatcher, adj)
+        sockinfo = (socket.AF_UNIX, socket.SOCK_STREAM, None, None)
+        return UnixWSGIServer(
+            application,
+            map,
+            _start,
+            _sock,
+            dispatcher=dispatcher,
+            adj=adj,
+            sockinfo=sockinfo)
+
+    effective_listen = []
+    last_serv = None
+    for sockinfo in adj.listen:
+        last_serv = TcpWSGIServer(
+            application,
+            map,
+            _start,
+            _sock,
+            dispatcher=dispatcher,
+            adj=adj,
+            sockinfo=sockinfo)
+        effective_listen.append((last_serv.effective_host, last_serv.effective_port))
+
+    # We are running a single server, so we can just return it...
+    if len(adj.listen) == 1:
+        return last_serv
+
+    return BaseServer(map, adj, effective_listen, dispatcher)
+
+class BaseServer(object):
```

Comment:
I can definitely add some more comments as to why i added this BaseServer, and rename it. I wasn't in love with the name, I just needed a name.

Technically I could have returned any of the BaseWSGIServer instances that were created and things would have worked. There is however no reason to keep a reference to them or pass them to BaseServer (ignore the name for now...) since the map defines what asyncore is going to act upon.

This piece of code simple exists to return something to the caller that has a `.run()` which starts the asyncore loop.